services/venice-server/src/main/java/com/linkedin/venice/listener/StatsHandler.java

```diff
@@ -273,12 +269,12 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
       if (!statCallbackExecuted) {
         recordBasicMetrics();
 
-        optionalKeySizeList.ifPresent(
-            keySizeList -> keySizeList.forEach(keySize -> currentStats.recordKeySizeInByte(storeName, keySize)));
-        optionalValueSizeList.ifPresent(
-            valueSizeList -> valueSizeList
-                .forEach(valueSize -> currentStats.recordValueSizeInByte(storeName, valueSize)));
-
+        if (keySizeList != null) {
+          keySizeList.forEach(keySize -> currentStats.recordKeySizeInByte(storeName, keySize));
```

Comment:
Another fun one... our usage of `currentStats` causes a number of map lookups roughly equal to ~20 + 2 * record count for batch gets / query... 

If we inverted `AggServerHttpRequestStats` (like we do in `HostLevelIngestionStats`) then we could look up the appropriate `ServerHttpRequestStats` once, and use it for every stat. This would eliminate all but one of these map lookups.